### PR TITLE
Migrate bounty payout address to BC beneficiary

### PR DESCRIPTION
**Pull request by Waldz**
Closes: #2956

This checks does free auto-registration to BC only if identity used to have payout address for bounty payments.
Also migrates found payout address to identity beneficiary in BC.

### CODE REVIEW

File: identity/registry/provider_registrar.go [hunk 1]
```diff
@@ -170,27 +186,41 @@ func (pr *ProviderRegistrar) handleEvent(qe queuedEvent) error {
...
+			log.Info().Msgf("provider %q not eligible for auto registration, will require manual registration", id.Address)
+			return nil
+		}
+	}
+
+	// This part migrates bounty payout address to BC beneficiary
+	payout, err := pr.mysteriumAPI.GetPayoutInfo(id, pr.signerFactory(id))
+	if err != nil {
+		if errHTTP, ok := err.(*requests.ErrorHTTP); ok && errHTTP.Code == 404 {
+			log.Info().Msgf("provider %q not have payout address for auto registration, will require manual registration", id.Address)
```
Comment (tomasmik) [inline identity/registry/provider_registrar.go:208]: Do we have to return here and not continue the registration? If I'm understanding this correctly new nodes will no longer auto register if we do this.

Comment (Waldz) [reply identity/registry/provider_registrar.go:208]: This is how i recognise newly created identity, which have to be on-boarded manually via UI

Comment (tomasmik) [reply identity/registry/provider_registrar.go:208]: Ye it seems like you can do that and it works. Nice

**Review by tomasmik — APPROVED** [review]

**Comment by chompomonim** [comment]
If I understand correctly, you expect that there will be input in api database about payout addresses. I think we may simply copy them from `testnet.api` into `testnet2.api`.

**Review by mdomasevicius — APPROVED** [review]

**Review by chompomonim — APPROVED** [review]

File: identity/registry/provider_registrar_test.go [hunk 2]
```diff
@@ -222,3 +227,16 @@ func (mt *mockTransactor) RegisterIdentity(id string, stake, fee *big.Int, benef
 func (mt *mockTransactor) CheckIfRegistrationBountyEligible(identity identity.Identity) (bool, error) {
```
Comment (chompomonim) [inline identity/registry/provider_registrar_test.go:227]: Had we already existing such function before (not mock)? Can't find one in this PR.

Comment (Waldz) [reply identity/registry/provider_registrar_test.go:227]: Yes we do - for free registration (with tokens i guess). But it's enabled for Betanet, but not Testnet2

**Review by soffokl — APPROVED** [review]

**Pull request merged**